modules/platform/video/nuxeo-platform-video/src/main/java/org/nuxeo/ecm/platform/video/service/RecomputeVideoConversionsAction.java

```diff
@@ -0,0 +1,181 @@
+                                              .collect(Collectors.toList());
+            }
+        }
+
+        @Override
+        protected void compute(CoreSession session, List<String> ids, Map<String, Serializable> properties) {
+            log.debug("Compute action: {} generating conversions: {} for doc ids: {}", ACTION_NAME, conversionNames,
+                    ids);
+            for (String docId : ids) {
+                if (!session.exists(new IdRef(docId))) {
+                    log.debug("Doc id doesn't exist: {}", docId);
+                    continue;
+                }
+
+                DocumentModel workingDocument = session.getDocument(new IdRef(docId));
+                Property fileProp = workingDocument.getProperty(xpath);
+                if (!((fileProp instanceof Blob) || (fileProp instanceof BlobProperty))) {
+                    log.warn("Property: {} of doc id: {} is not a blob.", xpath, docId);
+                    continue;
+                }
+                Blob blob = (Blob) fileProp.getValue();
+                if (blob == null) {
+                    // do nothing
+                    log.debug("No blob for doc: {}", workingDocument.getId());
+                    continue;
+                }
+                try {
+                    VideoDocument videoDoc = workingDocument.getAdapter(VideoDocument.class);
+                    Video video = videoDoc.getVideo();
+
+                    for (String conversion : conversionNames) {
+                        // here we want to commit the transaction since the video conversion can be very long
+                        TransactionHelper.commitOrRollbackTransaction();
+                        TranscodedVideo transcodedVideo = null;
+                        try {
+                            transcodedVideo = videoService.convert(video, conversion);
+                        } catch (ConversionException e) {
+                            log.warn("Conversion: {} of doc id: {} has failed", conversion, docId);
+                        }
+                        TransactionHelper.startTransaction();
+                        saveRendition(session, new IdRef(workingDocument.getId()), conversion, transcodedVideo);
+                    }
+                } catch (DocumentNotFoundException e) {
+                    // a parent of the document may have been deleted.
+                    continue;
+                } finally {
+                    if (!TransactionHelper.isTransactionActive()) {
+                        TransactionHelper.startTransaction();
+                    }
+                }
+                workingDocument.refresh();
+                DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), workingDocument);
+                Event event = ctx.newEvent(VIDEO_CONVERSIONS_DONE_EVENT);
+                Framework.getService(EventService.class).fireEvent(event);
+            }
+        }
+
+        @SuppressWarnings("unchecked")
```

Comment:
Please set the `@SuppressWarnings` on the smallest possible scope, which I guess is the `var transcodedVideos` definition bellow.